### PR TITLE
Cleanup usages of ExecutionEngine and QueryExecutionEngine

### PR DESCRIPTION
**Pull request by davidegrohmann**
Based on top of #6453

### CODE REVIEW

**Comment by pontusmelke** [comment]
@davidegrohmann it doesn't look like it figures out that the first couple of commits are already merged? Rebase?

**Comment by davidegrohmann** [comment]
@pontusmelke rebased

**Comment by pontusmelke** [comment]
@davidegrohmann looks like you need to rebase again or rather cherry pick the commits so we don't get all the commits appearing twice

File: community/kernel/src/main/java/org/neo4j/kernel/api/KernelAPI.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
      * Creates and returns a new {@link KernelTransaction} capable of modifying the
      * underlying graph.
      */
-    KernelTransaction newTransaction() throws TransactionFailureException;
+    KernelTransaction newTransaction( KernelTransaction.Type type ) throws TransactionFailureException;
```
Comment (spacecowboy) [inline community/kernel/src/main/java/org/neo4j/kernel/api/KernelAPI.java]: Can we add `@param` mention in javadoc?

File: community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java [hunk 2]
```diff
@@ -140,4 +145,12 @@
...
      */
     void registerCloseListener( CloseListener listener );
+
+    /**
+     * Kernel transaction type
+     *
+     * Implicit if created internally in the database
+     * Explicit if created by the end user
+     */
+    Type transactionType();
```
Comment (spacecowboy) [inline community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java]: Think we should have a `@return` note in the java doc here.

**Comment by spacecowboy** [comment]
So _implicit_ transactions are internally generated, and _explicit_ are user generated. But what is the actual difference between those? 

Will for example _explicit_ transactions be subject to security constraints and _internal_ not?

**Comment by davidegrohmann** [comment]
@spacecowboy explicit/implicit distinction is need to perform checks around periodic commit in Cypher due to the fact that the creation of transaction has been pulled out by Cypher.

**Comment by spacecowboy** [comment]
@davidegrohmann Ah I see. I have finished my comments btw.

**Comment by spacecowboy** [comment]
Typing here instead since all the rebases keep deleting my comments.

Still think `isPristine()` in `TransactionHandle.java` should be renamed to something more descriptive, such as `isImplicit()`.

No idea what a pristine transaction is.

**Comment by davidegrohmann** [comment]
@spacecowboy I have renamed it in #6515

**Comment by spacecowboy** [comment]
@davidegrohmann :+1: Then I have no further comments. Shall I do the merge honors when it goes green?

**Comment by davidegrohmann** [comment]
@spacecowboy yes please

**Comment by davidegrohmann** [comment]
@spacecowboy since #6515 is green and it is a superset of this, perhaps we can close this PR and simply merge the other one. WDYT?

**Comment by spacecowboy** [comment]
@davidegrohmann I think that is reasonable. You have to rebase it again though :(
I have assigned myself to the other PR.

**Pull request closed without merging**